### ISSUE TITLE
Annotations are not preserved on the proxy pod

### ISSUE THREAD

**Issue by gmcelhoe** [telepresenceio/telepresence#1430]
I noticed this because we use kube2iam to pass credentials to the pods, which looks at the pod's `iam.amazonaws.com/role` annotation.

The workaround is to use the `TELEPRESENCE_USE_DEPLOYMENT` setting to maintain the old deployment behavior.

I'm not sure of the ideal solution, there may need to be an "approved" or "blocked" list of annotations to be copied over to the proxy pod.  In our case, if all annotations were copied over, it would be best.

To detail our specific scenario, the original pod had the following annotations:
```
Annotations:    iam.amazonaws.com/role: arn:aws:iam::421435469981:role/aws-codecommit-read-only
                kubectl.kubernetes.io/restartedAt: 2020-07-20T23:09:19-04:00
                kubernetes.io/limit-ranger:
                  LimitRanger plugin set: cpu, ephemeral-storage, memory request for container monitor; cpu, ephemeral-storage, memory limit for container m...
                prometheus.io/path: /actuator/prometheus
                prometheus.io/port: 9081
                prometheus.io/scrape: false
```

and the proxy was reduced to only:

```
Annotations:  kubernetes.io/limit-ranger:
                LimitRanger plugin set: cpu, ephemeral-storage, memory request for container monitor; cpu, ephemeral-storage, memory limit for container m...
```

**Comment by ark3** [comment]
Thank you for the issue.

Notes for myself: The original Deployment has a [PodTemplateSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podtemplatespec-v1-core) that contains the metadata and the spec for the Pod. Telepresence is using the spec correctly but largely ignoring the metadata.

**Comment by ark3** [comment]
Fixed in #1434
